### PR TITLE
core,eth,les: update unclean shutdown markers regularly

### PR DESCRIPTION
**Pull request by s1na**
Successor to https://github.com/ethereum/go-ethereum/pull/22974, fixes https://github.com/ethereum/go-ethereum/issues/22580

### CODE REVIEW

File: eth/backend.go [hunk 1]
```diff
@@ -577,9 +583,31 @@ func (s *Ethereum) Stop() error {
 	s.miner.Close()
 	s.blockchain.Stop()
 	s.engine.Close()
-	rawdb.PopUncleanShutdownMarker(s.chainDb)
+	// Stop updating unclean shutdown markers and pop the last marker.
+	done := make(chan struct{})
+	s.shutdownCh <- done
+	<-done
```
Comment (holiman) [inline eth/backend.go:589]: No need to wait for `done`, just do a successfull send here and we're done

File: eth/backend.go [hunk 2]
```diff
@@ -577,9 +583,31 @@ func (s *Ethereum) Stop() error {
...
 	s.chainDb.Close()
 	s.eventMux.Stop()
 
 	return nil
 }
+
+func (s *Ethereum) updateUncleanShutdownMarkers() {
+	// update marker every five minutes
+	ticker := time.NewTicker(300 * time.Second)
+	defer func() { ticker.Stop() }()
```
Comment (rjl493456442) [inline eth/backend.go:600]: you can simplify the code by `defer ticker.Stop()`

**Review by holiman — APPROVED** [review]
LGTM

**Comment by s1na** [comment]
Updated based on the feedback I got on triage.

**Review by holiman — APPROVED** [review]
LGTM!

**Comment by fjl** [comment]
Package name 'node' is not great because it conflicts with the existing top-level package name we already have. Maybe call it 'shutdowncheck' or something. It doesn't need to be a good/short name.

**Comment by s1na** [comment]
I wanted to make it more general so we could put other modules in there too. But yeah the name conflict wasn't great. I renamed it.

**Comment by s1na** [comment]
Updated based on standup discussion. I could've avoided `MarkStartup` by having that logic in `New`, but somehow that didn't feel clean.

**Pull request merged**